src/Symfony/Component/Yaml/Parser.php

```diff
@@ -1183,60 +1187,99 @@ private function parseQuotedString(string $yaml): ?string
         return $value;
     }
 
-    private function lexInlineMapping(string $yaml): string
+    private function lexInlineQuotedString(int &$cursor): string
     {
-        if ('' === $yaml || '{' !== $yaml[0]) {
-            throw new \InvalidArgumentException(sprintf('"%s" is not a sequence.', $yaml));
-        }
-
-        for ($i = 1; isset($yaml[$i]) && '}' !== $yaml[$i]; ++$i) {
-        }
+        $quotedStringOffset = $cursor;
+        $cursor += strcspn($this->currentLine, $this->currentLine[$cursor], $cursor + 1) + 2;
```

Comment:
what's about escaping the delimiter ? `foo: "quote \"me\"..."`

Reply:
Not sure if you meant that, but escaped quotes are not yet handled here and are leading to problems for example in this yaml test:
```
[
    ["te\"st"],["test"]
]
```

A test case like `["te\"st"]` would still pass because the function just reads the inline sequence to the end of the document without finding the end of its assumed second quote. It doesn't report a problem though.
Another simple error case is:

``` 
[
    ["te\"st]"]
]
```

Reply:
Thank you both @jderusse and @simonberger for your careful review. I have updated the lexing of quoted strings to take your considerations into account.